script/translate.go

```diff
@@ -24,7 +24,9 @@ import (
 var trans = make(map[string]string)
 var attrRe = regexp.MustCompile(`\{\{\s*'([^']+)'\s+\|\s+translate\s*\}\}`)
 var attrReCond = regexp.MustCompile(`\{\{.+\s+\?\s+'([^']+)'\s+:\s+'([^']+)'\s+\|\s+translate\s*\}\}`)
-var jsRe = regexp.MustCompile(`\$translate.instant\("([^"]+)"\)`)
+
+// Find both `$translate.instant("…")` and `$translate.instant("…",…)` in JS.
+var jsRe = regexp.MustCompile(`\$translate.instant\("((.+?))"(,|\))`)
```

Comment:
Did you try my suggestion?  It would always look for the closing brace, even with the comma involved.

Reply:
I've changed the regex completely to catch all the different variants of `$translate.instant`. For testing, I cleared the English lang file completely and re-run the script from scratch, and it did find everything, i.e. the file came up exactly the same as before with no existing translations lost).

Please let me know if you think this should be simplified and/or done differently.

Reply:
I do find the construct with alternative double `"` or single `'` quotes rather fragile, because it may start with one and end with the other style.  Starting from my suggestion, I tried just using two compiled regex variants and looping through them.  Much simpler and less confusing with mixed quotes, and the result is the same as in your test. Note that escaped quotes (`' this isn\'t nice'` or `"let's be \"nice\" today"`) don't work well with the translations at all, hence can be ignored.

Here's my code:

```go
var jsRe = []*regexp.Regexp{
	regexp.MustCompile(`\$translate\.instant\("([^"]+?)"(,.*\)|\))`),
	regexp.MustCompile(`\$translate\.instant\('([^']+?)'(,.*\)|\))`),
}
```

And below where it's used:

```go
			for s := bufio.NewScanner(fd); s.Scan(); {
				for _, re := range jsRe {
					for _, matches := range re.FindAllStringSubmatch(s.Text(), -1) {
						translation(matches[1])
					}
				}
			}
```

Reply:
Done.